src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/filter.ts

```diff
@@ -39,3 +39,25 @@ export const filterWithLabelSchema = schema.object({
 });
 
 export type LensApiFilterType = typeof filterSchema.type;
+
+/**
+ * Unified search filter schema that can accept either a full filter object or a simple query string.
+ */
+export const unifiedSearchFilterSchema = schema.object({
```

Comment:
I had to implement the "complex" filter schema for now. Will likely swap with the shared schema proposed by presentation in a follow up